packages/ee/ui-platform/src/lib/pages/platform-settings/platform-settings.component.ts

```diff
@@ -1,71 +1,68 @@
-    {
-      type: 'TXT',
-      content: 'iXaWXthPkMSgNqfBaOezQGeioUgRpsui',
-      name: 'xyz.customdomain.co',
-    },
-    {
-      type: 'CNAME',
-      content: 'K5Rnd1OLw9L6HamjeZWdhrHWOwQ3DLaD.co',
-      name: 'xyz.customdomain.co',
-    },
-    {
-      type: 'TXT',
-      content: 'oOytttj9S3vGUyYsalUsVqQm7EzIonbl',
-      name: '_cf-customdomain.hostname.axzxcasd',
-    },
-  ];
-  platformId = '';
-  customDomainNote$: Observable<string>;
-  message = $localize`Please set the following TXT and CNAME records in your DNS provider, then click verify to confirm your control over the domain.`;
-  constructor(
-    private fb: FormBuilder,
-    private matSnakcbar: MatSnackBar,
-    private authenticationService: AuthenticationService
-  ) {
-    this.platformId = this.authenticationService.getPlatformId();
-    this.formGroup = this.fb.group({
-      customDomain: this.fb.control({
-        disabled: false,
-        value: 'xyz.customdomain.co',
-      }),
-    });
-    this.customDomainNote$ =
-      this.formGroup.controls.customDomain.valueChanges.pipe(
-        startWith(''),
-        map((res) => {
-          if (!res) {
-            return `Set up a CNAME for your custom domain, resolving to <b>abcdxyz.activepieces.co</b>, with the lowest TTL possible`;
-          } else {
-            return `Set up a CNAME for your <b>${res}</b>, resolving to <b>abcdxyz.activepieces.co</b>, with the lowest TTL possible`;
-          }
-        })
+  fragmentChanged$: Observable<string | null>;
+  readonly signingKeys = $localize`Signing Keys`;
+  readonly termsAndPrivacy = $localize`Privacy & Terms`;
+  readonly accountManagementEmail = $localize`Account Management Emails`;
+  readonly tabIndexFragmentMap: { [index: number]: string } = {
+    0: 'signingKeys',
+    1: 'SMTP',
+    2: 'TermsAndServices',
+  };
+  constructor(private router: Router, private route: ActivatedRoute) {
+    this.fragmentChanged$ = this.route.fragment.pipe(
+      tap((fragment) => {
+        if (fragment === null) {
+          this.updateFragment(this.tabIndexFragmentMap[0]);
+        } else {
+          this.fragmentCheck(fragment);
+        }
+      })
+    );
+  }
+  ngAfterViewInit(): void {
+    const fragment = this.route.snapshot.fragment;
+    if (fragment === null) {
+      this.updateFragment(this.tabIndexFragmentMap[0]);
+    } else {
+      this.fragmentCheck(fragment);
+    }
+  }
+
+  private fragmentCheck(fragment: string) {
+    if (this.tabGroup) {
+      const tabIndex = Object.values(this.tabIndexFragmentMap).indexOf(
+        fragment
       );
+      if (tabIndex >= 0) {
+        this.tabGroup.selectedIndex = tabIndex;
+      }
+    }
+  }
+
+  updateFragment(newFragment: string) {
+    this.router.navigate([], {
+      fragment: newFragment,
+    });
   }
-  copyValue(val: string) {
-    copyText(val);
-    this.matSnakcbar.open('Copied successfully');
+  tabChanged(event: MatTabChangeEvent) {
+    if (event.index < 0 || event.index > 2) return;
```

Comment:
what does that mean?

One way you can make it function with declerative name

IsSomething

Reply:
I think 0 and 2 is hardcarded? can they be changed?